From 68eef1713c73a8a2266c98e7ac36e822ad06cd7b Mon Sep 17 00:00:00 2001
From: Abhishek Chauhan <quic_abchauha@quicinc.com>
Date: Tue, 9 Nov 2021 15:42:41 -0800
Subject: [PATCH] hostap-2.9: Driver changes to set iface name

Driver changes to update macsec iface name  once wpa supplicant command is
executed.

Change-Id: Id59d03252995c9cc15df400a0770f0f350d7a3fe
Signed-off-by: Abhishek Chauhan <quic_abchauha@quicinc.com>
---
 src/drivers/driver_macsec_linux.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/drivers/driver_macsec_linux.c b/src/drivers/driver_macsec_linux.c
index a79a414..0ef69d8 100644
--- a/src/drivers/driver_macsec_linux.c
+++ b/src/drivers/driver_macsec_linux.c
@@ -1111,6 +1111,26 @@ nla_put_failure:
	return ret;
 }

+static int set_iface_name(struct transmit_sc *sc,
+		struct macsec_drv_data *drv,
+		struct rtnl_link *link)
+{
+	char ifname[20] = {0};
+	char pifname[20] = {0};
+	int i,j;
+	os_strlcpy(ifname, ".s0", sizeof(pifname));
+	os_strlcpy(pifname, drv->common.ifname, sizeof(pifname));
+	for(i=0; pifname[i]!='\0'; ++i);
+	for(j=0; ifname[j]!='\0'; ++j, ++i)
+	{
+		pifname[i]=ifname[j];
+	}
+	pifname[i]='\0';
+	/*So if parent iface is eth0 then link set is eth0.s0*/
+	rtnl_link_set_name(link, pifname);
+}
+
+
 /**
  * macsec_drv_create_transmit_sc - Create secure connection for transmit
  * @priv: private driver interface data from init()
@@ -1145,6 +1165,7 @@ static int macsec_drv_create_transmit_sc(
		return -1;
	}

+	set_iface_name(sc,drv,link);
	rtnl_link_set_link(link, drv->parent_ifi);

	sci = mka_sci_u64(&sc->sci);
--
2.7.4

